lib/MetaCPAN/Query/Release.pm

```diff
@@ -581,6 +581,16 @@ sub versions {
     my $data = [ map { $_->{fields} } @{ $ret->{hits}{hits} } ];
     single_valued_arrayref_to_scalar($data);
 
+    if ( $params->{versions} ) {
+        my %versions = map { $_ => undef } split /,/, $params->{versions};
+        $data = [ grep { exists $versions{ $_->{version} } } @{$data} ];
+    }
+
+    if ( $params->{plain} ) {
```

Comment:
I think it would make sense to put this part in the controller instead. It's weird to have a function like this return two entirely different forms of data.

Reply:
I agree about the plain mode - I'll make that change.

For the versions filtering I think I'll make it a filter in the ES query itself. I don't think that logic shouldn't be in the controller.

Reply:
Yeah, the plain mode part is what I meant. The version filtering belongs in the query module.